### ISSUE TITLE
Initial values of bi-directional variables are overwritten with 0 from the back channel

### ISSUE THREAD

**Issue by killenb** [ChimeraTK/ApplicationCore#113]
At least I see it in testable mode. The test uses `setScalarDefault()` to set the value, the modules which receives it does not write the back channel (but propagates the value to its outputs, which is working, `StateMachine/dcVoltageSetpoint` is OK), but I see 0 in the CS where I wrote the default value.

testSumLimiter in piezoctrl, lines 170 to 177
```
    testFacility.setScalarDefault<double>("/Channel0/VoltageSumLimiter/acVoltageSetpoint", 25.);
    testFacility.setScalarDefault<double>("/Channel0/VoltageSumLimiter/dcVoltageSetpoint", 10.);
    testFacility.runApplication();

    BOOST_CHECK_CLOSE(testFacility.readScalar<double>("/Channel0/StateMachine/dcVoltageSetpoint"), 10, 0.001);
    BOOST_CHECK_CLOSE(testFacility.readScalar<double>("/Channel0/StateMachine/acVoltageSetpoint"), 25, 0.001);
    BOOST_CHECK_CLOSE(testFacility.readScalar<double>("/Channel0/VoltageSumLimiter/dcVoltageSetpoint"), 10, 0.001);
    BOOST_CHECK_CLOSE(testFacility.readScalar<double>("/Channel0/VoltageSumLimiter/acVoltageSetpoint"), 25, 0.001);

```
Test result:
```
3: /home/killenb/piezoctrl/tests/testSumLimiter.cc(176): error in "testInitialisation": difference{100%} between testFacility.readScalar<double>("/Channel0/VoltageSumLimiter/dcVoltageSetpoint"){0} and 10{10} exceeds 0.001%
3: /home/killenb/piezoctrl/tests/testSumLimiter.cc(177): error in "testInitialisation": difference{100%} between testFacility.readScalar<double>("/Channel0/VoltageSumLimiter/acVoltageSetpoint"){0} and 25{25} exceeds 0.001%
```

**Comment by mhier** [comment]
The values seem to be overwritten only on the control system side. The ApplicationModule still sees the correct initial value.

This is caused by the propagation of the DataValidity flag after the devices get opened. At that point most likely the initial value has not yet been read by the ApplicationModule, so the writeAll() will write back a 0. This behaviour is presumably even subject to a race condition, since it depends whether the device was opened before the ApplicationModule accesses it or not.

The underlying problem is actually more conceptionally. Currently, writeAll(), readAll() and friends are always including the return channels of bidirectionals. I think this is wrong in many cases. Hence we should change the behaviour of these functions, which should also correct the problem with the initial values and return channels.

**Comment by mhier** [comment]
Actually there is another problem. The TestFacility writes the initial values "around" the TestableModeAccessorDecorator. Later reading back the value uses the decorator. If the readLatest() doesn't read a value (because no values was sent back through the return channel yet), the initial value of the decorator is still seen, which is 0.

**Comment by mhier** [comment]
Fixed with 098d6c65df26c518635752b7c129b8501d70c077 and 378a0c1833cfd806a3919962994ba3de37f56597
